core/blockchain.go

```diff
@@ -1891,7 +1892,7 @@ func (bc *BlockChain) initSnapshot(b *types.Header) {
 // it reaches a block with a state committed to the database. reprocessState does not use
 // snapshots since the disk layer for snapshots will most likely be above the last committed
 // state that reprocessing will start from.
-func (bc *BlockChain) reprocessState(current *types.Block, reexec uint64) error {
+func (bc *BlockChain) reprocessState(ctx context.Context, current *types.Block, reexec uint64) error {
```

Comment:
I really live the addition of an expiration context here. Two concerns:
1. In the case of a context expiration, should we clear out the `current` block so that this method would be atomic ?
2. Is there any state issue with partial execution of this method ? i.e. would terminating the execution in the middle would bring the node to some undesired state ?

Reply:
I think contexts used to gracefully shutdown are safe to use, but we decided not to merge this PR as contexts from VM init can be closed from parent thread when `SetState` is called.